client/src/root.js

```diff
@@ -67,29 +67,38 @@ class Root extends Component {
+		                    if (link && path.startsWith(link)) {
+		                        topLevelMenu.push(<NavLink key={entryKey.toLowerCase()} className="active" to={link}>{entry.title} <span className="sr-only">{t('current')}</span></NavLink>);
+		                    } else {
+		                        topLevelMenu.push(<NavLink key={entryKey.toLowerCase()} to={link}>{entry.title}</NavLink>);
+		                    }
+		                }
+		            }
 
-                if (mailtrainConfig.isAuthenticated) {
                     return (
                         <>
                             <ul className="navbar-nav mt-navbar-nav-left">
                                 {topLevelMenu}
                                 <NavDropdown label={t('administration')}>
-                                    {mailtrainConfig.globalPermissions.displayManageUsers && <DropdownLink to="/users">{t('users')}</DropdownLink>}
-                                    <DropdownLink to="/namespaces">{t('namespaces')}</DropdownLink>
-                                    {mailtrainConfig.globalPermissions.manageSettings && <DropdownLink to="/settings">{t('globalSettings')}</DropdownLink>}
-                                    <DropdownLink to="/send-configurations">{t('sendConfigurations')}</DropdownLink>
-                                    {mailtrainConfig.globalPermissions.manageBlacklist && <DropdownLink to="/blacklist">{t('blacklist')}</DropdownLink>}
-                                    <DropdownLink to="/account/api">{t('api')}</DropdownLink>
+                                    {(mailtrainConfig.user.admin || mailtrainConfig.globalPermissions.displayManageUsers) &&
+                                        <DropdownLink to="/users">{t('users')}</DropdownLink>}
+                                    {(mailtrainConfig.user.admin || mailtrainConfig.globalPermissions.displayManageNamespaces) &&
+                                        <DropdownLink to="/namespaces">{t('namespaces')}</DropdownLink>}
+                                    {(mailtrainConfig.user.admin || mailtrainConfig.globalPermissions.manageSettings) &&
+                                        <DropdownLink to="/settings">{t('globalSettings')}</DropdownLink>}
+                                    {(mailtrainConfig.user.admin || mailtrainConfig.globalPermissions.displayManageSendConfigurations) &&
+                                        <DropdownLink to="/send-configurations">{t('sendConfigurations')}</DropdownLink>}
+                                    {(mailtrainConfig.user.admin || mailtrainConfig.globalPermissions.manageBlacklist) &&
+                                        <DropdownLink to="/blacklist">{t('blacklist')}</DropdownLink>}
+                                    {(mailtrainConfig.user.admin || mailtrainConfig.globalPermissions.displayManageApi) &&
+                                        <DropdownLink to="/account/api">{t('api')}</DropdownLink>}
                                 </NavDropdown>
```

Comment:
Would you also mind adding a check that if none of the items under Administration are present, the top-level Administration label is also hidden?

Reply:
Ready ;)

Reply:
Lovely!